source/tutorials/hpc.rst

```diff
@@ -0,0 +1,688 @@
+and configure and enable the the munge service.
+
+#. Install |CL| on the controller node and add a user with adminstrator 
+   privilege.  
+
+#. Boot it up and log in.
+
+#. Update |CL| to the latest version.
+
+   .. code-block:: bash
+
+      sudo swupd update
+
+#. Set the password for the root user.
+
+   .. code-block:: bash
+
+      sudo passwd root
+
+#. Set the hostname of the system to `hpc-controller`.
+
+   .. code-block:: bash
+
+      sudo hostnamectl set-hostname hpc-controller
+
+#. Set the date and time to synchronize with an NTP server.
+
+   .. code-block:: bash
+
+      sudo timedatectl set-ntp true
+
+#. Install the `cluster-tools` bundle.
+
+   .. code-block:: bash
+
+      sudo swupd bundle-add cluster-tools
+
+#. Start the slurm controller service and enable it to start automatically 
+   on boot.
+
+   .. code-block:: bash
+
+      sudo systemctl enable slurmctld --now
+
+#. Create a munge key and start the munge service.
+   
+   a. Create the munge key.
+      
+      .. code-block:: bash
+         
+         sudo mkdir /etc/munge
+         dd if=/dev/urandom bs=1 count=1024 | sudo tee -a /etc/munge/munge.key
+
+   #. Set the ownership to `munge` and set the correct access permissions.
+     
+      .. code-block:: bash
+         
+         sudo chown munge: /etc/munge/munge.key
+         sudo chmod 400 /etc/munge/munge.key      
+
+   #. Start the munge service and set it to start automatically on boot.
+
+      .. code-block:: bash
+               
+         sudo systemctl enable munge --now
+
+#. Test munge.
+
+   a. Create a munge credential.
+
+      .. code-block:: bash
+
+         munge -n
+
+      Example output:
+
+      .. code-block:: console
+
+         MUNGE:AwQFAAC8QZHhL/+Fqhalhi+ZJBD5LavtMa8RMles1aPq7yuIZq3LtMmrB7KQZcQjG0qkFmoIIvixaCACFe1stLmF4VIg4Bg/7tilxteXHS940cuZ/TxpIuqC6fUH8zLgUZUPwJ4=:
+
+
+   #. Validate a munge credential.  
+      
+      .. code-block:: bash
+
+         munge -n | unmunge | grep STATUS
+
+      Example output:
+        
+      .. code-block:: console
+
+         STATUS:           Success (0)
+
+Set up worker nodes
+*******************
+
+For each worker node, perform these steps:
+
+#. Install |CL| on the worker node and add a user with adminstrator privilege.  
+   Ensure the username is the same as the one on the controller node.  This
+   is needed to simplify password-less-SSH-access setup in the next section.  
+
+#. Boot it up and log in.
+
+#. Update |CL| to the latest version.
+
+   .. code-block:: bash
+
+      sudo swupd update
+
+#. Set the password for the root user.
+
+   .. code-block:: bash
+
+      sudo passwd root
+
+#. Permit the root user to SSH.  This is necessary for password-less SSH setup 
+   later.
+
+   .. code-block:: bash
+
+      echo "PermitRootLogin yes" | sudo tee -a /etc/ssh/sshd_config 
+      sudo systemctl restart sshd
+
+#. Set the hostname to `hpc-worker` plus its number.
```

Comment:
"its number"? Where is its number found? Please be more specific and reference this number beforehand.

Reply:
As described at the intro section, this tutorial involves 4 worker nodes.  Their hostnames are "hpc-worker1", "hpc-worker2", "hpc-worker3", and "hpc-worker4".  Those are the numbers.

Reply:
Revision makes it much clearer. Thanks.